algoliasearch/index_iterator.go

```diff
@@ -23,14 +23,21 @@ func NewIndexIterator(index *Index, params map[string]interface{}) (it IndexIter
 func (it *IndexIterator) Next() (res map[string]interface{}, err error) {
 	// Abort if the user call `Next()` on a IndexIterator that has been
 	// initialized without being able to load the first page.
-	if it.page.NbHits == 0 {
-		err = fmt.Errorf("No more hits")
+	if len(it.page.Hits) == 0 {
+		err = errors.New("No more hits")
 		return
 	}
 
 	// If the last element of the page has been reached, the next one is loaded
```

Comment:
The main goal of this message is to differentiate a real error from the end of iteration, having a too long message isn't a good idea in my opinion